File: 程式設計：資料結構答案檔/C#/302.cs
using System;

class BubbleSortDemo
{
    static void Swap(ref int a, ref int b)
    {
        int temp = a;
        a = b;
        b = temp;
    }

    static void BubbleSort(int[] arr)
    {
        for (int i = 0; i < arr.Length; i++)
        {
            Console.Write(arr[i]);
            if (i < arr.Length - 1)
            {
                Console.Write(" ");
            }
        }
        Console.WriteLine();

        for (int i = 0; i < arr.Length; i++)
        {
            bool swapped = false;
            for (int j = 0; j < arr.Length - i - 1; j++)
            {
                if (arr[j] > arr[j + 1])
                {
                    Swap(ref arr[j], ref arr[j + 1]);
                    swapped = true;
                }
            }
            if (!swapped)
            {
                break;
            }

            for (int k = 0; k < arr.Length; k++)
            {
                Console.Write(arr[k]);
                if (k < arr.Length - 1)
                {
                    Console.Write(" ");
                }
            }
            Console.WriteLine();
        }
    }

    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        string[] inputs = Console.ReadLine().Split(' ');
        int[] numbers = new int[n];
        for (int i = 0; i < n; i++)
        {
            numbers[i] = int.Parse(inputs[i]);
        }

        BubbleSort(numbers);
    }
}

File: 程式設計：資料結構答案檔/C++/302.cpp
#include <stdio.h>

void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

void bubble_sort(int arr[], int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%d", arr[i]);
        if (i < n - 1)
        {
            printf(" ");
        }
    }
    printf("\n");

    for (int i = 0; i < n; i++)
    {
        int swapped = 0;
        for (int j = 0; j < n - i - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                swap(&arr[j], &arr[j + 1]);
                swapped = 1;
            }
        }
        if (!swapped)
        {
            break;
        }
        for (int k = 0; k < n; k++)
        {
            printf("%d", arr[k]);
            if (k < n - 1)
            {
                printf(" ");
            }
        }
        printf("\n");
    }
}

int main()
{
    int n;
    scanf("%d", &n);
    int input_numbers[n];
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &input_numbers[i]);
    }
    bubble_sort(input_numbers, n);

    return 0;
}
